File: rest-service/src/main/java/mb/lib/db/MBlastDBManager.java
package mb.lib.db;

import java.sql.Connection;

import org.veupathdb.lib.container.jaxrs.utils.db.DbManager;

public class MBlastDBManager implements AutoCloseable
{
  private final Connection con;

  public MBlastDBManager() throws Exception {
    con = DbManager.userDatabase().getDataSource().getConnection();
  }

  protected Connection getConnection() {
    return con;
  }

  @Override
  public void close() throws Exception {
    con.close();
  }
}

File: rest-service/src/main/kotlin/mb/lib/db/MBlastDBManager.kt
package mb.lib.db

import org.apache.logging.log4j.LogManager
import org.veupathdb.lib.container.jaxrs.utils.db.DbManager
import java.io.Closeable
import java.sql.Connection

open class MBlastDBManager(
  protected val connection: Connection = DbManager.userDatabase().dataSource.connection
): AutoCloseable, Closeable {

  private val Log = LogManager.getLogger(this::class.java)

  init {
    Log.debug("Opening connection {}", connection)
  }

  override fun close() {
    Log.debug("Closing connection {}", connection)
    connection.close()
  }
}
